### DISCUSSION TITLE
Getting revert Statement when coding along in Chapter 4

### DISCUSSION THREAD

**Discussion by infinite969** [Cyfrin/foundry-full-course-cu#747] [Q&A] [answered]
Hello friends, I am getting this revert statement when calling my "GetVersion" function with Patrick in chapter 4 at timestamp 4:41:34 of part 1 on youtube.
```
**Here is the revert statement:**

"revert
	The transaction has been reverted to the initial state.
Note: The called function should be payable if you send value and the value you send should be less than your current balance.
Debug the transaction to get more information."

**Here is my code so far:**
```
// SPDX-License-Identifier: MIT
pragma solidity 0.8.18;
  import "AggregatorV3Interface.sol";

// ** Code instuctions **
// Recieve donation from users in ETH
// Withdraw donations 
// Keep a list of donation addresses 

// ETH/USD PRICE FEED ADDRESS: 0x694AA1769357215DE4FAC081bf1f309aDC325306

contract FundMe {
    
    uint256 public miminumUSD  = 5;
    
    function Donate() public payable {
        require (msg.value > 1e18,"didnt send enough eth");
    }

    function GetPrice() public {
        
    }

    function GetVersion() public view returns (uint256)  {
        return AggregatorV3Interface(0x694AA1769357215DE4FAC081bf1f309aDC325306).version();
    
    }
    
    
  //function GetConversionRate() public {}
}
```
thx in advance ❤️

**Comment by infinite969** [comment]
hmm seems that posted in chat kinda weird 😂

**Comment by infinite969** [comment]
For more context I saved the Chainlink v3 interface in my solidity as opposed to importing from Github as well

**Comment by theirrationalone** [comment, marked as answer]
First of all, Please Update Your codebase like this...

```solidity
// SPDX-License-Identifier: MIT

pragma solidity 0.8.18;
import "@chainlink/contracts/src/v0.8/interfaces/AggregatorV3Interface.sol";

// ** Code instuctions **
// Recieve donation from users in ETH
// Withdraw donations
// Keep a list of donation addresses

// ETH/USD PRICE FEED ADDRESS: 0x694AA1769357215DE4FAC081bf1f309aDC325306

contract FundMe {

    uint256 public miminumUSD  = 5;

    function Donate() public payable {
        require (msg.value > 1e18,"didnt send enough eth");
    }

    function GetVersion() public view returns (uint256)  {
        return AggregatorV3Interface(0x694AA1769357215DE4FAC081bf1f309aDC325306).version();
    }
}
```

Before Deploying your contract, Please connect remix with your metamask....

Follow image below...

![Screenshot (129)](https://github.com/Cyfrin/foundry-full-course-f23/assets/96755510/86921de2-14e7-40a9-a557-46965a8b0abb)

**Reply by infinite969** [reply]
I see. I was using the remix merge vm I'm guessing that's why it was not able to interact with the chainlink v3 interface being monitored on the sepolia test network. Is this correct? The revert message did not appear when calling the "GetVersion" function on the contract deployed to the test network. Thank you for your help

**Reply by theirrationalone** [reply]
You're Welcome❤
Yes Right catch 🔥
